addons/payment_authorize/controllers/main.py

```diff
@@ -38,13 +36,15 @@ def authorize_s2s_create_json(self, **kwargs):
         acquirer = request.env['payment.acquirer'].browse(acquirer_id)
         if not kwargs.get('partner_id'):
             kwargs = dict(kwargs, partner_id=request.env.user.partner_id.id)
-        return acquirer.s2s_process(kwargs).id
+        try:
+           return acquirer.s2s_process(kwargs).id
+        except:
```

Comment:
You may catch exception just like on the route below and return a dict with the error message.
This allows the customer to see exactly what error has been triggered instead of a generic message.